src/matcher_combinators/core.clj

```diff
@@ -49,6 +49,22 @@
      :else
      nil)))
 
+(defn- regex? [value] (instance? java.util.regex.Pattern value))
+(defrecord Regex [expected]
+  Matcher
+  (match [_this actual]
+   (if-let [issue (validate-input expected actual regex? (constantly true) 'regex "java.util.regex.Pattern")]
+     issue
+     (try
+       (if (re-find expected actual)
```

Comment:
have you considered using `re-matches`? I don't know if there is any difference in performance
(this more of a bike sheeding question)

Reply:
🤔 is the output of these functions used when they match?
Because when they don't match `re-find` just returns `nil` right?